pkg/sql/catalog/descpb/structured.proto

```diff
@@ -980,7 +980,9 @@ message TableDescriptor {
   // as a table. The data on disk is refreshed with the REFRESH MATERIALIZED
   // VIEW command. This flag is only set when ViewQuery != "".
   optional bool is_materialized_view = 41 [(gogoproto.nullable) = false];
-
+  // RefreshViewRequired indicates if the materialized view needs to be refreshed
+  // prior to access.
+  optional bool refresh_view_required = 53 [(gogoproto.nullable) = false];
```

Comment:
I looked a bit into the feasibility of checking for the presence of table statistics instead of maintaining a field in the descriptor, but it doesn't seem super straightforward to do, plus having a field such as this one doesn't add much technical debt. Perhaps that's what we'll move towards in the long run. Materialized views don't exactly have a lot of polish to them, such as they are, but that's OK for the time being too.

How do you feel though about making the field a `uint64` counter instead or a boolean flag? You could the increment that counter at every refresh, instead of setting the flag to false. It feels more useful to have a counter, somehow.